test/integration/migrate/index.js

```diff
@@ -268,12 +221,52 @@ module.exports = function(knex) {
 
   });
 
+  describe('knex.migrate.latest in parallel', function() {
+
+    afterEach(function() {
+      return knex.migrate.rollback({directory: 'test/integration/migrate/test'});
+    });
+
+    if (knex.client.dialect === 'postgres') {
+      it("is able to run two migrations in parallel in postgres", function () {
```

Comment:
Sorry, you're right. I guess I got confused with the whole auto-commit thing.

However, the reason concurrent migrations usually **cannot succeed** in MySQL, MariaDB and Oracle is because DDL statements (e.g. `CREATE TABLE` that is used in the test) cause an implicit commit to the current transaction. See https://dev.mysql.com/doc/refman/5.7/en/implicit-commit.html.

This is what makes the test fail: once the first table in the first migration is created, the transaction is implicitly committed and the awaiting `select for update` statement returns with a value of `1` (i.e. locked) - which causes knex to throw. I've turned on knex query logging where it is seen clearly:
```
 knex:query BEGIN; +0ms
 knex:bindings undefined +0ms
 knex:query BEGIN; +1ms
 knex:bindings undefined +0ms
 knex:query select * from `knex_migrations_lock` for update +2ms
 knex:bindings [] +0ms
 knex:query select * from `knex_migrations_lock` for update +2ms
 knex:bindings [] +0ms
 knex:query update `knex_migrations_lock` set `is_locked` = ? +1ms
 knex:bindings [ 1 ] +0ms
 knex:query show tables like ? +2ms
 knex:bindings [ 'knex_migrations' ] +0ms
 knex:query show tables like ? +1ms
 knex:bindings [ 'knex_migrations_lock' ] +0ms
 knex:query select * from `knex_migrations_lock` +2ms
 knex:bindings [] +0ms
 knex:query select `name` from `knex_migrations` order by `id` asc +1ms
 knex:bindings [] +0ms
 knex:query select max(`batch`) as `max_batch` from `knex_migrations` +2ms
 knex:bindings [] +0ms
 knex:query create table `migration_test_1` (`id` int unsigned not null auto_increment primary key, `name` varchar(255)) default character set utf8 +2ms
 knex:bindings [] +0ms
 Knex:warning - Can't take lock to run migrations: Migration table is already locked
 Knex:warning - If you are sure migrations are not running you can release the lock manually by deleting all the rows from migrations lock table: knex_migrations_lock
 knex:query ROLLBACK; +8ms
```

Unlike the others, in Postgres most DDL statements [do not cause implicit commits](http://stackoverflow.com/a/4736346/2295972), therefore the awaiting transaction will continue only after the lock is released.

I am using Postgres and for my use case this PR would help a great deal, since it is the cleanest way to avoid collisions during concurrent migrations, so I think it is worth it.